persistence/src/androidMain/kotlin/com/wire/kalium/persistence/db/UserDatabaseProvider.kt

```diff
@@ -1,217 +0,0 @@
-import androidx.sqlite.db.SupportSQLiteDatabase
-import app.cash.sqldelight.EnumColumnAdapter
-import app.cash.sqldelight.adapter.primitive.IntColumnAdapter
-import app.cash.sqldelight.driver.android.AndroidSqliteDriver
-import com.wire.kalium.persistence.Call
-import com.wire.kalium.persistence.Client
-import com.wire.kalium.persistence.Connection
-import com.wire.kalium.persistence.Conversation
-import com.wire.kalium.persistence.Member
-import com.wire.kalium.persistence.Message
-import com.wire.kalium.persistence.MessageAssetContent
-import com.wire.kalium.persistence.MessageConversationChangedContent
-import com.wire.kalium.persistence.MessageFailedToDecryptContent
-import com.wire.kalium.persistence.MessageMemberChangeContent
-import com.wire.kalium.persistence.MessageMention
-import com.wire.kalium.persistence.MessageMissedCallContent
-import com.wire.kalium.persistence.MessageRestrictedAssetContent
-import com.wire.kalium.persistence.MessageTextContent
-import com.wire.kalium.persistence.MessageUnknownContent
-import com.wire.kalium.persistence.User
-import com.wire.kalium.persistence.UserDatabase
-import com.wire.kalium.persistence.cache.LRUCache
-import com.wire.kalium.persistence.dao.BotServiceAdapter
-import com.wire.kalium.persistence.dao.ConnectionDAO
-import com.wire.kalium.persistence.dao.ConnectionDAOImpl
-import com.wire.kalium.persistence.dao.ContentTypeAdapter
-import com.wire.kalium.persistence.dao.ConversationAccessListAdapter
-import com.wire.kalium.persistence.dao.ConversationAccessRoleListAdapter
-import com.wire.kalium.persistence.dao.ConversationDAO
-import com.wire.kalium.persistence.dao.ConversationDAOImpl
-import com.wire.kalium.persistence.dao.MemberRoleAdapter
-import com.wire.kalium.persistence.dao.MetadataDAO
-import com.wire.kalium.persistence.dao.MetadataDAOImpl
-import com.wire.kalium.persistence.dao.QualifiedIDAdapter
-import com.wire.kalium.persistence.dao.QualifiedIDListAdapter
-import com.wire.kalium.persistence.dao.TeamDAO
-import com.wire.kalium.persistence.dao.TeamDAOImpl
-import com.wire.kalium.persistence.dao.UserDAO
-import com.wire.kalium.persistence.dao.UserDAOImpl
-import com.wire.kalium.persistence.dao.UserIDEntity
-import com.wire.kalium.persistence.dao.asset.AssetDAO
-import com.wire.kalium.persistence.dao.asset.AssetDAOImpl
-import com.wire.kalium.persistence.dao.call.CallDAO
-import com.wire.kalium.persistence.dao.call.CallDAOImpl
-import com.wire.kalium.persistence.dao.client.ClientDAO
-import com.wire.kalium.persistence.dao.client.ClientDAOImpl
-import com.wire.kalium.persistence.dao.message.MessageDAO
-import com.wire.kalium.persistence.dao.message.MessageDAOImpl
-import com.wire.kalium.persistence.util.FileNameUtil
-import kotlinx.coroutines.CoroutineDispatcher
-import kotlinx.coroutines.CoroutineScope
-import kotlinx.coroutines.SupervisorJob
-import kotlinx.coroutines.cancel
-import net.sqlcipher.database.SupportFactory
-
-actual class UserDatabaseProvider(
```

Comment:
🙌

Reply:
I'm proud of this one 😅
It deserved its own PR to be honest. But it was something I identified during the work in reactions and I just had to do ASAP because I couldn't just add more repeated logic for each platform code.

I think this "Function as a Constructor" is a strong pattern we can safely use in KMP projects.

I've seen it multiple times in some open source libraries but took me some time to wrap my head around and find a place where we could use it.